src/h5web/visualizations/nexus/utils.ts

```diff
@@ -30,13 +31,52 @@ export function getAttributeValue(
   return entity.attributes.find((attr) => attr.name === attributeName)?.value;
 }
 
+export function getChildEntityByName(
+  group: HDF5Group,
+  metadata: HDF5Metadata,
+  entityName: string | undefined
+): HDF5Entity | undefined {
+  if (!entityName) {
+    return undefined;
+  }
+
+  const childLink = group.links?.find((l) => l.title === entityName);
+
+  return getEntity(childLink, metadata);
+}
+
 export function getSignalDataset(
   group: HDF5Group,
   metadata: HDF5Metadata
 ): HDF5Dataset | undefined {
   const signal = getAttributeValue(group, 'signal');
-  const signalLink = signal && group.links?.find((l) => l.title === signal);
+  const signalDataset = getChildEntityByName(group, metadata, signal as string);
 
-  const signalDataset = getEntity(signalLink, metadata);
   return signalDataset && isDataset(signalDataset) ? signalDataset : undefined;
 }
+
+export function getAxesDatasets(
+  group: HDF5Group,
+  metadata: HDF5Metadata
+): (HDF5Dataset | undefined)[] | undefined {
+  const axesList = getAttributeValue(group, 'axes') as
+    | string
+    | Array<string>
+    | undefined;
+  const axes =
+    typeof axesList === 'string'
+      ? (axesList as string).match(/\w+|\./g)
+      : axesList;
```

Comment:
Parse the `axes` attributes. 

I consider two usecases for now:
- The attribute is an array of strings (ex: `['x', 'y']). In this case, no parsing is needed.
- The attribute is a string with axis names separated by a separator character (ex: `"x y"`) as shown in [NeXuS docs](https://manual.nexusformat.org/rules.html?highlight=axes). To convert in an array, I match with a simple reg exp. If no match, this will be `null`. The particular case `.` (no dataset) is matched by the regex as well which will lead to an `undefined` dataset below.

Reply:
Alright, so this can all be simplified as instructed by Armando and Thomas.

Reply:
True. The NeXuS page I linked is in fact outdated: https://github.com/nexusformat/NIAC/issues/76 

 I consider now the following possibilities:

- The attribute is an array of strings (ex: `['x', 'y']`). 
- The attribute is a string. In this case, I consider the string as a single axis value (ex: `"time"` is interpreted as `["time"]`)